scripts/mobile-latency-trace/analyze.py

```
@@ -0,0 +1,744 @@
+    for event in ios.get("ev.grid", []):
+        decode_us = event.integer("dec_us")
+        if decode_us is not None:
+            metrics.setdefault("iOS: ev.grid decode", []).append(decode_us / 1_000.0)
+        sequence = event.integer("seq")
+        surface = event.fields.get("s")
+        if sequence is None or surface is None:
+            continue
+        gate = first_after(
+            ios.get("gate", []),
+            event.time_us,
+            lambda stamp: stamp.fields.get("s") == surface
```

Ruff (0.16.0)

[warning] Function definition does not bind loop variable `surface`

(B023)